cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -175,7 +163,7 @@ func (c *cachingMigInfoProvider) findMigWithMatchingBasename(instanceRef GceRef)
 }
 
 func (c *cachingMigInfoProvider) fillMigInstances(migRef GceRef) error {
-	if val, ok := c.migInstancesLastRefreshedInfo[migRef.String()]; ok {
+	if val, ok := c.cache.GetMigInstancesRefreshTime(migRef); ok {
```

Comment:
Since the scope of the PR is to fix concurrent access to the map `c.migInstancesLastRefreshedInfo`, this comment could be considered out of scope.

The current implementation of the `fillMigInstances` doesn't respect `c.migInstancesMinRefreshWaitTime` since concurrent calls can fetch the last refresh time and all calls can consider the data stale and make concurrent `c.gceClient.FetchMigInstances()` calls. This points a bug in the underlying implementation of RefreshWaitTime which is only enforced in linear calls this func but not concurrent.

Reply:
I think this is a very good point, but a bit outside of the intended scope of this PR. We can improve on this mechanism further in a separate change.